alvr/xtask/src/ci.rs

```diff
@@ -0,0 +1,87 @@
+use serde::Deserialize;
+use serde_json::{self as json, Deserializer, Value};
+use xshell::{cmd, Shell};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+enum Level {
+    Error,
+    Warning,
+    Note,
+    Help,
+    FailureNote,
+}
+
+#[derive(Debug, Deserialize)]
+struct Span {
+    file_name: String,
+    line_start: u64,
+    line_end: u64,
+    column_start: u64,
+    column_end: u64,
+    is_primary: bool,
+}
+
+// https://doc.rust-lang.org/rustc/json.html
+#[derive(Debug, Deserialize)]
+struct CompilerMessage {
+    // message_type
+    level: Level,
+    spans: Vec<Span>,
+    rendered: String,
+}
+
+pub fn clippy_ci() {
```

Comment:
Are you sure there is no better way to pretty print the output? The cargo check gh action was supposed to do that.

Reply:
That gh action is long abandoned and using that api is horrible because if the pr doesn't satisfy very specific conditions it's not getting the api key and then the annotations aren't being sent. Only remaining mechanisms are using these workflow commands or having problem matchers (ginormous regex to parse compiler output), and I think this is the much cleaner solution.